extensions/ql-vscode/src/run-queries.ts

```diff
@@ -516,67 +552,78 @@ export async function compileAndRunQueryAgainstDatabase(
   }
 
   const query = new QueryInfo(qlProgram, db, packConfig.dbscheme, quickEvalPosition, metadata, templates);
-  await checkDbschemeCompatibility(cliServer, qs, query, progress, token);
 
-  let errors;
+  const upgradeDir = tmp.dirSync({ dir: upgradesTmpDir.name });
```

Comment:
Avoid using sync functions in user code. You can use the `tmp-promise` package to make an async call to creating a  tmp directory. After changing the import statement, all you need to do is this:

```suggestion
  const upgradeDir = await tmp.dir({ dir: upgradesTmpDir.name });
```